### ISSUE TITLE
[request] Improvements to dataset code for pretraining

### ISSUE THREAD

**Issue by patricklabatut** [facebookresearch/dinov2#243]
- Only depend on standard ImageNet files
- Facilitate swapping in a different dataset

Related issues:
- #233
- #126
- #100
- #72

**Comment by BrianPulfer** [comment]
Here's a [labels.txt](https://github.com/facebookresearch/dinov2/files/13427966/labels.txt) file that works out of the box. Only "issue" with this is that commas in the class name are removed. If it is of interest, I can open a PR :)

**Comment by ayushnangia** [comment]
so I might be a bit dumb but how to just pretrain the old weights of image net with my custom dataset. or do i need to train from scratch with my custom dataset @patricklabatut

**Comment by wyh196646** [comment]
> so I might be a bit dumb but how to just pretrain the old weights of image net with my custom dataset. or do i need to train from scratch with my custom dataset @patricklabatut

I have same problem，and there is a tutorial for training my own dataset via dinov2 self supervised learning method？

**Comment by xinli2008** [comment]
@wyh196646 have you soloved the problem? pretraining dinov2 on my own dataset without label

**Comment by 1921134176** [comment]
A simple approach is that you can organize your data like an image classification task, where the data root contains a series of category subdirectories under which the images you want to train on are located. Then configure the dataset type inside to be changed to ImageFolder, e.g. dataset_path: ImageFolder:root=/ssd/cx_dataset/flower_photos. then in the loaders.py script from torchvision.datasets import ImageFolder and add elif name == “ImageFolder”: class_ = ImageFolder on line 60.

**Comment by divyanshj16** [comment]
Did anyone create their custom dataset?
I have 2M images. I want to do further pretraining on this dataset.

**Comment by junlinguo** [comment]
> Did anyone create their custom dataset? I have 2M images. I want to do further pretraining on this dataset.

Hi! I used 3M images (in npy format, 8 bands) and just put images into one folder. I customized the dataset class so it can just read the images in a folder. You can check out github (in-progress). I provide a readme file for how to adjust for customized dataset. You probably just need to modify the image format, i assume you used png or jpg, etc. 

https://github.com/geopacha/DeepAndes/blob/main/docs/adjust_pretrain_for_8bands.md#1-dataset-module

the corresponding module: 
https://github.com/geopacha/DeepAndes/blob/main/dinov2_ssl_8bands/dinov2/data/datasets/nlb_dataset.py


follow our work / DM if you may have any help from us
@article{guo2025deepandes,
  title={DeepAndes: A Self-Supervised Vision Foundation Model for Multi-Spectral Remote Sensing Imagery of the Andes},
  author={Guo, Junlin and Zimmer-Dauphinee, James R and Nieusma, Jordan M and Lu, Siqi and Liu, Quan and Deng, Ruining and Cui, Can and Yue, Jialin and Lin, Yizhe and Yao, Tianyuan and others},
  journal={arXiv preprint arXiv:2504.20303},
  year={2025}
}
